tutorials/interactive/index.md

```diff
@@ -0,0 +1,814 @@
+has previously been favorited. The number next to the star indicates that a
+total of 41 people have favorited this lake.  After completing this tutorial,
+tapping the star removes its favorited status, replacing
+the solid star with an outline, and decreasing the count. Tapping
+again favorites the lake, drawing a solid star and increasing the count.
+
+The following shows both states.
+
+<img src="images/favorited-not-favorited.png" alt="the custom widget you'll create">
+
+To accomplish this, you'll create a single custom widget that includes both the
+star and and the count. Tapping the star changes state for both widgets,
+so the same widget should manage both.
+
+If you don't already have the starting code for the Lakes example,
+get it from the links in the blue [**Note**](#note) box.
+
+### Contents
+
+* [Stateful and stateless widgets](#stateful-stateless)
+* [Creating a stateful widget](#creating-stateful-widget)
+  * [Step 1: Decide who manages the widget's state](#step-1)
+  * [Step 2: Subclass StatefulWidget](#step-2)
+  * [Step 3: Subclass the State object](#step-3)
+  * [Step 4: Plug the stateful widget into the widget tree](#step-4)
+  * [Problems?](#problems)
+* [Managing state](#managing-state)
+  * [The widget manages its own state](#self-managed)
+  * [The parent manages the widget's state](#parent-managed)
+  * [A mix-and-match approach](#mix-and-match)
+* [Other interactive widgets](#other-interactive-widgets)
+  * [Standard widgets](#standard-widgets)
+  * [Material widgets](#material-widgets)
+* [Resources](#resources)
+
+<a name="stateful-stateless"></a>
+## Stateful and stateless widgets
+
+<div class="panel" markdown="1">
+
+<b> <a id="whats-the-point" class="anchor" href="#whats-the-point" aria-hidden="true"><span class="octicon octicon-link"></span></a>What's the point?</b>
+
+* All widgets are either stateful or stateless.
+* _State_ means information about a widget that might change.
+  A checkbox's state, for example, would probably
+  include a boolean indicating whether it's currently checked.
+* If a widget changes based on user interaction, or as the result of a
+  data feed (for example), it's _stateful_.
+* All widgets should be _immutable_, meaning once created they cannot change.
+* Being immutable is fine for a stateless widget, because its appearance
+  never changes.
+* When a stateful widget changes (its slider moved, for example),
+  the widget needs to rebuild in order to update the UI.
+* A _State_ object holds the changeable information for a stateful widget,
+  separating the widget's state from its appearance.
+* The class managing the state object tells the widget to update the UI by
+  calling `setState`.
+* Typically, the state object is managed by the widget itself, the parent
+  widget, or some combination.
+* How the widget is implemented determines who manages its state.
+
+</div>
+
+There are two kinds of widgets, stateful and stateless.
```

Comment:
There are at least two other kinds of widgets, inheritedwidget and renderobjectwidget. There could also be others.